app/controllers/courses_controller.rb

```diff
@@ -105,8 +105,10 @@ def save_deadlines
 
     groups = group_params
     groups.each do |name, deadlines|
-      json_array = [deadlines[:static], deadlines[:unlock]].to_json
-      @course.exercise_group_by_name(name).group_deadline=(json_array)
+      soft_deadlines = [deadlines[:soft][:static], deadlines[:soft][:unlock]].to_json
+      hard_deadlines = [deadlines[:hard][:static], deadlines[:hard][:unlock]].to_json
```

Comment:
When we pass the input to the exercises, they try to parse it into a spec. If it fails, they raise an invalidsyntaxerror which is caught at the end of this method.